manifest: Update Matter SDK revision to pull new crypto changes (#13540)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst

Matter
------

> * From |NCS| 2.6.0 version the Arm PSA Crypto API is used as cryptography for devices that use Matter over Thread.

Reviewer: ```suggestion
* For devices that use Matter over Thread, the default cryptography backend is now Arm PSA Crypto API instead of Mbed TLS, which was used in earlier versions.
```

>   You can still build all examples with deprecated mbedTLS support by setting the :kconfig:option:`CONFIG_CHIP_CRYPTO_PSA` Kconfig option to ``n``, but you must build Thread library from sources.

Reviewer: ```suggestion
  You can still build all examples with deprecated Mbed TLS support by setting the :kconfig:option:`CONFIG_CHIP_CRYPTO_PSA` Kconfig option to ``n``, but you must build the Thread libraries from sources.
```

>   To use :ref:`inheritance of Thread certification <ug_matter_device_certification_reqs_dependent>` from Nordic Semiconductor you MUST use the PSA Crypto API backend.

Reviewer: ```suggestion
  To :ref:`inherit Thread certification <ug_matter_device_certification_reqs_dependent>` from Nordic Semiconductor, you must use the PSA Crypto API backend.
```

* Updated the page about :ref:`ug_matter_device_low_power_configuration` with the information about Intermittently Connected Devices (ICD) configuration.

* Added:

  * A Kconfig option for disabling or enabling :ref:`ug_matter_configuring_read_client`.
  * Migration of the Device Attestation Certificate (DAC) private key from the factory data set to the PSA ITS secure storage.

    The DAC private key can be removed from the factory data set after the migration.
    You can enable this experimental functionality by setting the :kconfig:option:`CONFIG_CHIP_CRYPTO_PSA_MIGRATE_DAC_PRIV_KEY` Kconfig option to ``y``.

>   * Migration of the Node Operational Key Pair (NOK) from the mbedTLS-related settings storage to the PSA ITS secure storage.

Reviewer: ```suggestion

  * Migration of the Node Operational Key Pair (NOK) from the Mbed TLS-related settings storage to the PSA ITS secure storage.
```

>     All existing NOKs (for all existing Matter fabrics) will be migrated to the PSA ITS secure storage at the boot of the device.

Reviewer: ```suggestion
    All existing NOKs for all Matter fabrics will be migrated to the PSA ITS secure storage at boot.

```
Since only a device boots, it's not necessary to say it's the boot of the device.

>     If the migration fails the ``kOperationalKeystoreMigrationFailed`` Matter's public event will be emitted and dispatched in the ``Matter event handler``.

Reviewer: ```suggestion
    If the migration fails, the ``kOperationalKeystoreMigrationFailed`` public event will be emitted and dispatched in the Matter event handler.
```

    The default reaction to migration failure is a factory reset of the device.

>     To change the default reaction set the :kconfig:option:`CONFIG_NCS_SAMPLE_MATTER_FACTORY_RESET_ON_KEY_MIGRATION_FAILURE` Kconfig option to `n` and implement the reaction in your ``Matter event handler``

Reviewer: ```suggestion
    To change the default reaction, set the :kconfig:option:`CONFIG_NCS_SAMPLE_MATTER_FACTORY_RESET_ON_KEY_MIGRATION_FAILURE` Kconfig option to ``n`` and implement the reaction in your Matter event handler.
```

>     For more information about the ``Matter event handler`` see the :file:`matter_event_handler.h` file.

Reviewer: ```suggestion
    For more information about the Matter event handler, see the :file:`matter_event_handler.h` header file.
```

>     After the migration, mbedTLS-related entries in the settings storage will be removed and not available anymore.

Reviewer: ```suggestion
    After the migration, Mbed TLS-related entries in the settings storage will be removed and no longer available.
```

------

* Updated the page about :ref:`ug_matter_device_low_power_configuration` with the information about Intermittently Connected Devices (ICD) configuration.

* Added:

  * A Kconfig option for disabling or enabling :ref:`ug_matter_configuring_read_client`.
  * Support for PSA Crypto API for devices that use Matter over Thread.
    It is enabled by default and can be disabled by setting the :kconfig:option:`CONFIG_CHIP_CRYPTO_PSA` Kconfig option to ``n``.
  * Migration of the Node Operational Key Pair (NOK) from the mbedTLS-related settings storage to the PSA ITS secure storage.
    All existing NOKs (for all existing Matter fabrics) will be migrated to the PSA ITS secure storage at the boot of the device.

>     After the migration mbedTLS-related entries in the settings storage will be removed and not available anymore.

Reviewer: ```suggestion
    After the migration, mbedTLS-related entries in the settings storage will be removed and not available anymore.
```

    The generated measurements simulate local temperature changes.
    Additionally, you can enable periodic outdoor temperature measurements by binding the thermostat with an external temperature sensor device.

  * Migration of the Node Operational Key Pair (NOK) from the Mbed TLS-related settings storage to the PSA ITS secure storage.
    All existing NOKs for all Matter fabrics will be migrated to the PSA ITS secure storage at boot.

    If the migration fails, the ``kOperationalKeystoreMigrationFailed`` public event will be emitted and dispatched in the Matter event handler.
    The default reaction to migration failure is a factory reset of the device.
    To change the default reaction, set the :kconfig:option:`CONFIG_NCS_SAMPLE_MATTER_FACTORY_RESET_ON_KEY_MIGRATION_FAILURE` Kconfig option to ``n`` and implement the reaction in your Matter event handler.
    For more information about the Matter event handler, see the :file:`matter_event_handler.h` header file.

>     After the migration, Mbed TLS-related entries in the settings storage will be removed and no longer available.

Reviewer: Line 129 should probably be moved to be after Line 123, as Lines 125 - 128 concern migration failure. Remember to keep the empty line to maintain the paragraph break.
